[ADD] barcode: GS1 nomenclature (#4352)
odoo/documentation: content/applications/inventory_and_mrp/inventory/barcode/operations/gs1_nomenclature.rst

Then, in the code block below, replace the second trigger parameter, "INSERT BARCODE HERE", with the
full barcode pattern, including the :abbr:`A.I. (application identifier)` (e.g. `01` for product
GTINs). Copy the full code block with a valid GS1 barcode and paste it into the console.

.. code-block:: javascript

   odoo.__DEBUG__.services['web.core'].bus.trigger(
      'barcode_scanned',
      "INSERT BARCODE HERE",
      $(".o_web_client")[0],
   )

.. example::
   The product, `Large Cabinet` is being received in the warehouse. The barcode is configured as
   follows:

   - Product identifier uses the :abbr:`A.I. (application identifier)` `01` + the 14-digit product
     barcode (including check digit) `33778856214558`.

   - Specify the quantity of products `30` + the quantity `00000001`.
   - Batch number uses the :abbr:`A.I. (application identifier)` `10` + `BATCH00000015`

   So, in :menuselection:`Barcode --> Operations --> Receipts`, open a receipt for the product and
   open the console. Paste the code block with the configured barcode in required location.

   .. code-block:: javascript

      odoo.__DEBUG__.services['web.core'].bus.trigger(
         'barcode_scanned',

>          "0133778856214558300000000110BATCH00000015",

Author: Add check in Lots/Serial Numbers page in Inventory.

         $(".o_web_client")[0],
      )

   .. image:: gs1_nomenclature/paste-in-console.png
      :align: center
      :alt: Paste code block into console

   Select the :guilabel:`Pencil icon` to view the details of the scan.

>    .. image:: gs1_nomenclature/open-pencil-icon.png

Reviewer: Can we add a image with the Batch Number set as well on the product?
```suggestion
   .. image:: gs1_nomenclature/open-pencil-icon.png
```

Next, select :menuselection:`Barcode Nomenclature --> Default GS1 Nomenclature` from the default
barcode nomenclature options.

.. image:: gs1_nomenclature/setup-gs1-nomenclature.png
   :align: center
   :alt: Choose GS1 from dropdown and click the internal link to see the list of GS1 rules.

To view and edit a list of GS1 *rules* and *barcode patterns* Odoo supports by default, click the
:guilabel:`➡️ (External link)` icon to the right of the :guilabel:`Barcode Nomenclature` selection.

Opening the pop-up table provides an editable view of GS1 :guilabel:`Rule Names` available in Odoo.
The table contains all the information that can be condensed with a GS1 barcode, along with the
corresponding :guilabel:`Barcode Pattern`.

.. tip::
   After setting GS1 as the barcode nomenclature, :menuselection:`Barcode Nomenclatures` can also be
   accessed by first enabling :ref:`developer mode <developer-mode>`. Navigate to
   :menuselection:`Inventory app --> Configuration --> Barcode Nomenclatures` and finally, select
   :guilabel:`Default GS1 Nomenclature`.

.. _barcode/operations/create-GS1-barcode:

> Use GS1 barcode

Reviewer 2: Suggestions to be more specific:

1.
```suggestion
Use GS1 barcodes in Odoo
```

2.
```suggestion
Use GS1 in Odoo Barcode
```

===============

> To build GS1 barcodes in Odoo, combine multiple pieces of information using the specified barcode

Reviewer 2: > To build GS1 barcodes in Odoo

This implies Odoo will build the barcodes, meaning, we give Odoo some input data, and Odoo will spit out output data in the form of barcodes.

That doesn't appear to be what's happening? Or at least...that's not clear to me in the text.

It seems like there are rules/conventions to follow, and that's fine, however, I don't see clear instructions or a clear conclusion that Odoo builds barcodes in this section...

pattern. The `application identifier
<https://www.gs1.org/standards/barcodes/application-identifiers>`_ (A.I.) serves as the universal
prefix for GS1 for barcode identification. Odoo uses regular expressions to describe barcode
patterns concisely. Each barcode pattern begins with a required 2-4 digit :abbr:`A.I. (application
identifier)`, which corresponds to the rule defined in the system's :ref:`barcode nomenclature list
<barcode/operations/set-up-barcode-nomenclature>`. By including the appropriate :abbr:`A.I.

> (application identifier)` from the list, Odoo can accurately interpret GS1 barcodes. While most

Reviewer 2: capitalize
```suggestion
(Application Identifier)` from the list, Odoo can accurately interpret GS1 barcodes. While most
```

+============+==========================+======+==================================+==========================================+
| Product    | Global Trade Item Number | 01   | (01)(\\d{14})                    | :guilabel:`Barcode` field on product form|
|            | (GTIN)                   |      |                                  |                                          |
+------------+--------------------------+------+----------------------------------+------------------------------------------+
| Quantity   | Variable count of items  | 30   | (30)(\\d{0,8})                   | :guilabel:`Units` field on transfer form |
+------------+--------------------------+------+----------------------------------+------------------------------------------+
| Lot Number | Batch or lot number      | 10   | (10)([!"%-/0-9:-?A-Z_a-z]{0,20}) | :guilabel:`Lot` on Detailed Operations   |
|            |                          |      |                                  | pop-up                                   |
+------------+--------------------------+------+----------------------------------+------------------------------------------+

.. _barcode/operations/lot-setup:

> Configuration

Reviewer 2: be more specific 😉 
also why is a **Configuration** heading arriving midway through this section (isn't it usually first?)

~~~~~~~~~~~~~

To track products using lots, first enable the :ref:`Lots and Serial Numbers
<inventory/management/track_products_by_lots>` feature. To do so, navigate to
:menuselection:`Inventory app --> Configuration --> Settings`. Next, under the
:guilabel:`Traceability` heading, check the box for :guilabel:`Lots & Serial Numbers`.

Then, set up the product barcode by navigating to the intended product form in
:menuselection:`Inventory app --> Products --> Products` and selecting the product. On the product
form, click :guilabel:`Edit`. Then, in the :guilabel:`General Information` tab, fill in the
:guilabel:`Barcode` field with the 14-digit `Global Trade Item Number (GTIN)
<https://www.gs1.org/standards/get-barcodes>`_, which is a universal and unique identifying number
from GS1.

.. important::
   On the product form, omit the :abbr:`A.I. (application identifier)` `01` for GTIN product barcode
   pattern, as it is only used to encode multiple barcodes into a single barcode that contains
   detailed information about the package contents.

.. example::

>    To create a barcode for the product, `Fuji Apple`, enter the 14-digit GTIN `12345678901231` in

Reviewer 2: this isn't helpful because this is demonstrates exactly what *not* to do 😓 

GTINs in particular follow very specific formatting rules, yes, however you can't just make them up (even if you follow the rules for how they're made, such as making sure the character count is right)— these numbers represent uniquely recognizable products in the global supply-chain and therefore have unique IDs. Here's [a table, for example,](https://resources.markon.com/product-catalog/apples-fuji) that shows a bunch of *actual* Fuji Apple GTIN/UPC values.

Would expand the example much more in the next iteration...as it reads right now it communicates "you can just make numbers up and put them here on the form"

   the :guilabel:`Barcode` field on the product form.

   .. image:: gs1_nomenclature/barcode-field.png
      :align: center
      :alt: Enter 14-digit GTIN into the Barcode field on product form.

.. tip::

>    It is also possible to view a list of all products and barcodes. To access this list, go to

Reviewer 2: wording.
```suggestion
   To view a list of *all* products and their corresponding barcodes in the Odoo database, navigate to...
```

| Quantity   | Variable count of items  | 30   | (30)(\\d{0,8})               | :guilabel:`Units` field on transfer form |
+------------+--------------------------+------+------------------------------+------------------------------------------+
| Lot Number | Batch or lot number      | 10   | (10)                         | :guilabel:`Lot` on Detailed Operations   |
|            |                          |      | ([!"%-/0-9:-?A-Z_a-z]{0,20}) | pop-up                                   |
+------------+--------------------------+------+------------------------------+------------------------------------------+

.. _barcode/gs1_nomenclature/lot-setup:

Configuration
~~~~~~~~~~~~~

To track products using lots, first enable the :ref:`Lots and Serial Numbers
<inventory/management/track_products_by_lots>` feature. To do so, navigate to
:menuselection:`Inventory app --> Configuration --> Settings`. Next, under the
:guilabel:`Traceability` heading, check the box for :guilabel:`Lots & Serial Numbers`.

Then, set up the product barcode by navigating to the intended product form in
:menuselection:`Inventory --> Products --> Products` and selecting the product. On the product form,

> click :guilabel:`Edit`. Then, in the :guilabel:`General Information` tab, fill in the

Author: Wow thanks for noticing this!

numbers, letters, special characters, and character length– to ensure proper interpretation of the
barcode. By scanning the barcode on an unopened box, GS1 nomenclature can identify the product, lot
number, number of units contained, and more.

.. seealso::
   - `All GS1 barcodes <https://www.gs1.org/standards/barcodes/application-identifiers>`_
   - :ref:`Odoo's default GS1 rules <barcode/default-gs1-nomenclature-list>`
   - :ref:`Why's my barcode not working? <barcode/operations/troubleshooting>`

.. _barcode/set-up-barcode-nomenclature:

Set up barcode nomenclature
===========================

To use GS1 nomenclature, navigate to the :menuselection:`Inventory app --> Configuration -->
Settings`. Then under the :guilabel:`Barcode` section, check the :guilabel:`Barcode Scanner` box.
Next, select :menuselection:`Barcode Nomenclature --> Default GS1 Nomenclature` from the default
barcode nomenclature options.

.. image:: gs1_nomenclature/setup-gs1-nomenclature.png
   :align: center
   :alt: Choose GS1 from dropdown and click the internal link to see the list of GS1 rules.

To view and edit a list of GS1 *rules* and *barcode patterns* Odoo supports by default, click the

> :guilabel:`➡️ (External link)` icon to the right of the :guilabel:`Barcode Nomenclature` selection.

Author: I wasn't able to find a more suitable emoji. Do you think the arrow is close enough, or should I just describe it a the "sideways arrow out of a box" icon?

Settings`. Then under the :guilabel:`Barcode` section, check the :guilabel:`Barcode Scanner` box.
Next, select :menuselection:`Barcode Nomenclature --> Default GS1 Nomenclature` from the default
barcode nomenclature options.

.. image:: gs1_nomenclature/setup-gs1-nomenclature.png
   :align: center
   :alt: Choose GS1 from dropdown and click the internal link to see the list of GS1 rules.

To view and edit a list of GS1 *rules* and *barcode patterns* Odoo supports by default, click the
:guilabel:`➡️ (External link)` icon to the right of the :guilabel:`Barcode Nomenclature` selection.

Opening the pop-up table provides an editable view of of GS1 :guilabel:`Rule Names` available in
Odoo. The table contains all the information that can be condensed with a GS1 barcode, along with
the corresponding :guilabel:`Barcode Pattern`.

.. tip::
   After setting GS1 as the barcode nomenclature, :menuselection:`Barcode Nomenclatures` can also be
   accessed by first enabling developer mode. Navigate to :menuselection:`Inventory app -->
   Configuration --> Barcode Nomenclatures` and finally, select :guilabel:`Default GS1
   Nomenclature`.

.. _barcode/create-GS1-barcode:

> Create GS1 barcode

Author: Good thinking, Melanie!
```suggestion
Use GS1 barcode
```

==================

To build GS1 barcodes in Odoo, combine multiple pieces of information using the specified barcode

> pattern. The `application identifier <barcode/operations/app-id>`_ (A.I.) serves as the universal

Author: fixed!! thanks for testing these links (:

+------------+--------------------------+------+------------------------------+------------------------------+
| Lot Number | Batch or lot number      | 10   | (10)                         | Lot on Detailed Operations   |
|            |                          |      | ([!"%-/0-9:-?A-Z_a-z]{0,20}) | pop-up                       |
+------------+--------------------------+------+------------------------------+------------------------------+

.. _barcode/gs1_nomenclature/lot-setup:

Configuration
~~~~~~~~~~~~~

To track products using lots, first enable the :ref:`Lots and Serial Numbers
<inventory/management/track_products_by_lots>` feature. To do so, navigate to
:menuselection:`Inventory app --> Configuration --> Settings`. Next, under the
:guilabel:`Traceability` heading, check the box for :guilabel:`Lots & Serial Numbers`.

Then, set up the product barcode by navigating to the intended product form in
:menuselection:`Inventory --> Products --> Products` and select the product. On the product form,
click :guilabel:`Edit`. Then, in the :guilabel:`General Information` tab, fill in the

> :guilabel:`Barcode` field with the 14-digit Global Trade Item Number (GTIN) of the product.

Author: ```suggestion
:guilabel:`Barcode` field with the 14-digit `Global Trade Item Number (GTIN) <https://www.gs1.org/standards/get-barcodes>`_, which is a universal and unique identifying number from GS1. 
```

      :align: center
      :alt: View the Product Barcodes page from inventory settings.

.. _barcode/gs1_nomenclature/lot-setup-on-product:

Next, enable lots and serial number tracking on the product. Select the :guilabel:`Inventory` tab on
the product form. Under :guilabel:`Tracking`, choose the :guilabel:`By Lots` radio button.

.. image:: gs1_nomenclature/track-by-lots.png
   :align: center
   :alt: Enable product tracking by lots in the "Inventory" tab of the product form.

Scan barcode on receipt
~~~~~~~~~~~~~~~~~~~~~~~

To ensure accurate lot interpretation in Odoo on product barcodes scanned during a receipt
operation, navigate to the :menuselection:`Barcode` app to manage the :ref:`receipt picking process
<barcode/operations/scan-received-products>`.

From the :guilabel:`Barcode Scanning` dashboard, click the :guilabel:`Operations` button, then the
:guilabel:`Receipts` button to view list of vendor receptions to process. Receipts generated from
abbr:`POs (Purchase Orders)` are listed, but new receipt operations can also be created directly
through the :menuselection:`Barcode` app using the :guilabel:`Create` button. Click on the

> warehouse operation (`WH/IN`) and scan product barcodes and lot numbers. Use the :guilabel:`✏️

Author: ```suggestion
warehouse operation (`WH/IN`) and scan product barcodes and lot numbers with a barcode scanner. The scanned product is then displayed on the mobile view. Use the :guilabel:`✏️
```

> (pencil)` button to designate the quantities for specific lot numbers.

Author: ```suggestion
(pencil)` button to designate the quantities for specific lot numbers. Clicking the pencil button opens a window,

In the edit quantities window, use the number pad and :guilabel:`⬅️ (delete arrow)` button to enter the exact quantity of items in the lot.
```

+------------+--------------------------+----------+-------------------+-------------------+
|    Name    |        Rule Name         |   A.I.   |  Barcode Pattern  |   Field in Odoo   |
+============+==========================+==========+===================+===================+
| Product    | Global Trade Item Number | 01       | (01)(\\d{14})     | Barcode field on  |
|            | (GTIN)                   |          |                   | product form      |
+------------+--------------------------+----------+-------------------+-------------------+
| Quantity in| Variable count of items  | 310[0-5] | (310[0-5])(\\d{6})| Units field on    |
| kilograms  |                          |          |                   | transfer form     |
+------------+--------------------------+----------+-------------------+-------------------+

Scan barcode on receipt
~~~~~~~~~~~~~~~~~~~~~~~

To confirm that quantities are correctly interpreted in Odoo, place an order in the *Purchase* app
using the appropriate unit of measure (:guilabel:`UoM`) for the quantity of products to be
purchased.

.. seealso::
   :ref:`Simplify vendor unit conversions with UoMs<inventory/products/uom-example>`

> After the order is placed, navigate to the :menuselection:`Barcode` app to receive the :ref:`vendor

Reviewer 3: The link is to "Receipts" so I think it makes sense for "receive" to be included in the link name 
```suggestion
After the order is placed, navigate to the :menuselection:`Barcode` app to :ref:`receive the vendor
```

shipment <barcode/operations/scan-received-products>`.

.. example::
   On the receipt in the *Barcode* app, receive an order for `52.1 kg` of peaches by scanning the
   barcode. If `52.1 / 52.1` :guilabel:`kg` appears on the page, this means the reception was
   processed without issue. Finally, press :guilabel:`Validate`.

   Note: the :abbr:`A.I. (application identifier)` for kilograms, `310` + `1`, was used to represent
   `52.1` kg as a barcode: `000521`. This is because the `1` represents how many digits from the

>    right to place the decimal.

Reviewer 3: ```suggestion
   right to place the decimal point.
```

   fields in the barcode are interpreted correctly.

Begin by turning on :ref:`developer mode <developer-mode>` and navigating to the :guilabel:`Barcode
Nomenclatures` list in :menuselection:`Inventory app --> Configuration --> Barcode Nomenclatures`.
Then, select the :guilabel:`Default GS1 Nomenclature` list item.

On the :guilabel:`Default GS1 Nomenclature` page, select :guilabel:`Add a line` at the bottom of the
table, which opens a window to create a new rule. The :guilabel:`Rule Name` field is used internally
to identify what the barcode represents. The barcode :guilabel:`Types` are different classifications
of information that can be understood by the system (e.g. product, quantity, best before date,
package, coupon). The :guilabel:`Sequence` represents the priority of the rule; this means the
smaller the value, the higher the rule appears on the table. Odoo follows the sequential order of
this table and will use the first rule it matches based on the sequence. The :guilabel:`Barcode
Pattern` is how the sequence of letters or numbers is recognized by the system to contain
information about the product.

> After filling the information, click the :guilabel:`Save & New` button to make another rule or click

Reviewer 3: ```suggestion
After filling in the information, click the :guilabel:`Save & New` button to make another rule or click
```

:guilabel:`Save & Close` to save and return to the table of rules.

.. _barcode/operations/troubleshooting:

Barcode troubleshooting
=======================

Since GS1 barcodes are challenging to work with, here are some checks to try when the barcodes are
not working as expected:

> #. The :guilabel:`Barcode Nomenclature` setting is set as :menuselection:`Default GS1 Nomenclature`. Jump to

Reviewer 3: ```suggestion
#. Ensure that the :guilabel:`Barcode Nomenclature` setting is set as :menuselection:`Default GS1 Nomenclature`. Jump to
```

| Count of trade items            | Quantity    | (37)(\\d{0,8})               | Measure            | Qty in units for   |
|                                 |             |                              |                    | containers (AI 02) |
+---------------------------------+-------------+------------------------------+--------------------+--------------------+
| Net weight: kilograms (kg)      | Quantity    | (310[0-5])(\\d{6})           | Measure            | Qty in kg          |
+---------------------------------+-------------+------------------------------+--------------------+--------------------+
| Length in meters (m)            | Quantity    | (311[0-5])(\\d{6})           | Measure            | Qty in m           |
+---------------------------------+-------------+------------------------------+--------------------+--------------------+
| Net volume: liters (L)          | Quantity    | (315[0-5])(\\d{6})           | Measure            | Qty in L           |
+---------------------------------+-------------+------------------------------+--------------------+--------------------+

> | Net volume: cubic meters (m^3)  | Quantity    | (316[0-5])(\\d{6})           | Measure            | Qty in m^3         |

Reviewer 3: If there's a way to format the exponents in LaTeX (so they look like exponents instead of `m^3`) that would be great!

Author: Did it :heart:

   After placing a :abbr:`PO (Purchase Order)` for 50 apples, navigate to the associated receipt.
   Scan the product barcode, and Odoo will prompt for the lot number.

   .. image:: gs1_nomenclature/receive-50-apples.png
      :align: center
      :alt: Scan the barcode for a product on the reception picking page in the *Barcode* app.

   Scan the lot number to process 1 of 50 apples. To avoid scanning 49 remaining barcodes, click
   the :guilabel:`✏️ (pencil)` button next to the desired lot number.

   .. image:: gs1_nomenclature/scan-apple-lot-number.png
      :align: center
      :alt: Scan lot number and click the pencil to edit quantities.

   Doing so opens a mobile-friendly keypad page to specify received quantities. Use the keypad to
   specify the :guilabel:`Units` for the lot number. When finished, click :guilabel:`Confirm`.

   .. image:: gs1_nomenclature/edit-lot-quantities.png
      :align: center
      :alt: Change scanned quantities using pencil button.

Repeat this process to specify additional lot numbers and quantities in this receipt. Once the
:guilabel:`Units` are all accounted for, finish the reception by clicking the :guilabel:`Validate`
button.

> .. note::
>    THIS SECTION WILL BE DELETED BEFORE PUBLISHING! It's for the reviewers to test these barcodes
>    easily.
> 
>    product: (01)12345678901231
>    qty: (30)00000050
>    Lot#1: (10)LOT0001
>    Lot#2: (10)LOT0002
> 
>    .. code-block:: javascript
> 
>       // First scan: product
>       odoo.__DEBUG__.services['web.core'].bus.trigger(
>          'barcode_scanned',
>          "0112345678901231",
>          $(".o_web_client")[0],
>       )
>       // Second scan: first Lot#, then enter qty with pencil button
>       odoo.__DEBUG__.services['web.core'].bus.trigger(
>          'barcode_scanned',
>          "10LOT0001",
>          $(".o_web_client")[0],
>       )
>       // Third scan: second Lot#, then enter qty.
>       odoo.__DEBUG__.services['web.core'].bus.trigger(
>          'barcode_scanned',
>          "10LOT0002",
>          $(".o_web_client")[0],
>       )

Author: ```suggestion

```

Scan barcode on receipt
~~~~~~~~~~~~~~~~~~~~~~~

To confirm that quantities are correctly interpreted in Odoo, place an order in the *Purchase* app
using the appropriate unit of measure (:guilabel:`UoM`) for the quantity of products to be
purchased.

.. seealso::
   :ref:`Simplify vendor unit conversions with UoMs<inventory/products/uom-example>`

After the order is placed, navigate to the :menuselection:`Barcode` app to :ref:`receive the vendor
shipment <barcode/operations/scan-received-products>`.

.. example::
   On the receipt in the *Barcode* app, receive an order for `52.1 kg` of peaches by scanning the
   barcode. If `52.1 / 52.1` :guilabel:`kg` appears on the page, this means the reception was
   processed without issue. Finally, press :guilabel:`Validate`.

   Note: the :abbr:`A.I. (application identifier)` for kilograms, `310` + `1`, was used to represent
   `52.1` kg as a barcode: `000521`. This is because the `1` represents how many digits from the
   right to place the decimal point.

   .. image:: gs1_nomenclature/scan-barcode-peaches.png
      :align: center
      :alt: Scan barcode screen for a reception operation in the Barcode app.

> .. note::
>    THIS SECTION WILL BE DELETED BEFORE PUBLISHING! It's for the reviewers to test these barcodes
>    easily!
> 
>    On product form for peach, in :guilabel:`General Information` tab, the :guilabel:`Barcode` field
>    =  `00614141000012`
> 
>    For kg, A.I : 310
>    decimal number: 1
>    value (6 digits): 000521
> 
>    .. code-block:: javascript
> 
>       odoo.__DEBUG__.services['web.core'].bus.trigger(
>          'barcode_scanned',
>          "01006141410000123101000521",
>          $(".o_web_client")[0],
>       )

Author: ```suggestion

```

For additional verification purposes, the quantities of received products are also recorded on the
:guilabel:`Product Moves` report, accessible by navigating to :menuselection:`Inventory app -->
Reporting --> Product Moves`.

The items on the :guilabel:`Product Moves` report are grouped by product by default. To confirm the
received quantities, click on a product line to open its collapsible drop-down menu, which displays
a list of *stock move lines* for the product. The latest stock move matches the warehouse reception
reference number (e.g. `WH/IN/00013`) and quantity processed in the barcode scan, demonstrating that
the records processed in the *Barcode* app were properly stored in *Inventory*.

.. image:: gs1_nomenclature/stock-moves-peach.png
   :align: center
   :alt: Reception stock move record for 52.1 kg of peaches.

.. _barcode/create-new-rules:

Create rules
------------

If a supplier uses a GS1 barcode with a field not supported by Odoo's :ref:`default GS1
list <barcode/default-gs1-nomenclature-list>`, Odoo will fail to interpret the entire barcode. To
ensure the complete reading of the barcode, it is necessary to add the missing barcode to Odoo's
list.

> .. warning::

Author: ```suggestion
.. important::
```

| Count of trade items                   | Quantity    | (37)(\\d{0,8})               | Measure            | Qty in units for   |
|                                        |             |                              |                    | containers (AI 02) |
+----------------------------------------+-------------+------------------------------+--------------------+--------------------+
| Net weight: kilograms (kg)             | Quantity    | (310[0-5])(\\d{6})           | Measure            | Qty in kg          |
+----------------------------------------+-------------+------------------------------+--------------------+--------------------+
| Length in meters (m)                   | Quantity    | (311[0-5])(\\d{6})           | Measure            | Qty in m           |
+----------------------------------------+-------------+------------------------------+--------------------+--------------------+
| Net volume: liters (L)                 | Quantity    | (315[0-5])(\\d{6})           | Measure            | Qty in L           |
+----------------------------------------+-------------+------------------------------+--------------------+--------------------+

> | Net volume: cubic meters (:math:`m^3`) | Quantity    | (316[0-5])(\\d{6})           | Measure            | Qty in :math:`m^3` |

Reviewer 3: The exponents in LaTex are super cool! They're looking a little out of place though, since the `math` formatting changes the font. But I found this `superscript` formatting that I think works perfectly!
```
m\ :sup:`3`
```
Using the `math` formatting for just the exponent could also work, although the exponent gets written in the math font:
```
m\ :math:`^3`
```

Author: Yayyy!! this workss

+----------------------------------------+-------------+------------------------------+--------------------+--------------------+
| Length in inches (in)                  | Quantity    | (321[0-5])(\\d{6})           | Measure            | Qty in inches      |
+----------------------------------------+-------------+------------------------------+--------------------+--------------------+
| Net weight/volume: ounces (oz)         | Quantity    | (357[0-5])(\\d{6})           | Measure            | Qty in oz          |
+----------------------------------------+-------------+------------------------------+--------------------+--------------------+

> | Net volume: cubic feet (:math:`ft^3`)  | Quantity    | (365[0-5])(\\d{6})           | Measure            | Qty in :math:`ft^3`|

Reviewer 3: Please change these to use the :sup: formatting as well :)
